cargo/lib/dependabot/cargo/update_checker/latest_version_finder.rb

```diff
@@ -52,19 +55,29 @@ def fetch_lowest_security_fix_version
           versions.min
         end
 
+        sig { params(versions_array: T::Array[Gem::Version]).returns(T::Array[Gem::Version]) }
         def filter_prerelease_versions(versions_array)
           return versions_array if wants_prerelease?
 
-          versions_array.reject(&:prerelease?)
+          filtered = versions_array.reject(&:prerelease?)
```

Comment:
just wondering; is this something that can be raised higher in the inheritance hierarchy so we don't have to implement it in every downstream implementation?

Reply:
Yes there are a few things like this and also filtering out versions that are lower than the current version that we could potentially pull up into the common updater code.

Reply:
And if that's prohibitively expensive or not feasible, please can we fix it for all ecosystems in one go so we don't have it linger :)

Reply:
It's definitely out of scope for this PR, I can file an issue to track that work.